src/mito2/src/worker/handle_alter.rs

```diff
@@ -120,11 +140,31 @@ impl<S> RegionWorkerLoop<S> {
             }
         };
         // Persist the metadata to region's manifest.
-        let change = RegionChange {
-            metadata: new_meta.clone(),
-        };
+        let change = RegionChange { metadata: new_meta };
         self.handle_manifest_region_change(region, change, sender)
     }
+
+    /// Handles requests that changes region options, like TTL. It only affects memory state
+    /// since changes are persisted in the `DatanodeTableValue` in metasrv.
+    fn handle_alter_region_options(
+        &mut self,
+        region: MitoRegionRef,
+        version: VersionRef,
+        options: Vec<ChangeRegionOption>,
+        sender: OptionOutputTx,
+    ) {
+        let mut current_options = version.options.clone();
+        for option in options {
+            match option {
+                ChangeRegionOption::TTL(new_ttl) => {
+                    info!("update ttl to {:?}", new_ttl);
```

Comment:
We should also log the region id.

Reply:
fixed in https://github.com/GreptimeTeam/greptimedb/pull/4848/commits/56f4be4c66137553ad3261fd1163e44d9939bdf7